core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -130,22 +132,25 @@ public void execute() throws CommandExecutionException {
             deleteFile(testPlanYAMLFilePathLocation);
             deleteParentDirectoryIfEmpty(testPlanYAMLFilePathLocation);
 
+            // Create infrastructure from config
             Infrastructure infrastructure = testConfig.getInfrastructure();
-            infrastructure.setInfraParams(testConfig.getInfraParams());
-            TestPlan testPlan = generateTestPlan(testConfig, scenarioRepoDir, infraRepo);
+            infrastructure.setInfraParams(testConfig.getInfraParams().get(0));
 
-            DeploymentPattern deploymentPattern = new DeploymentPattern();
-            deploymentPattern.setName(testConfig.getDeploymentPatterns().get(0));
-            deploymentPattern.setProduct(product);
-            deploymentPattern = persistDeploymentPattern(deploymentPattern);
-            testPlan.setDeploymentPattern(deploymentPattern);
-            testPlan.setInfraParameters(infrastructure.getInfraParams().toString());
+            // Create or get deployment pattern
+            DeploymentPattern deploymentPattern = getDeploymentPattern(product,
+                    testConfig.getDeploymentPatterns().get(0));
 
-            //Set logger file path
-            setLogFilePath(product, testPlan);
+            // Generate test plan from config
+            TestPlan testPlan = generateTestPlan(deploymentPattern, testConfig, scenarioRepoDir, infraRepo);
+
+            //Set the log file path
+            LogFilePathLookup.setLogFilePath(deriveLogFilePath(product, testPlan));
```

Comment:
Why are you modifying a static field here? This does not seem correct.

Reply:
This path is accessed via the log4j framework via the log4j.xml. We do only know the log file path in runtime and therefore need to modify the path in runtime.

Reply:
Setting the path at runtime is fine. But doing that in a static method is not correct. Anyway, this will be addressed via [1] as discussed. 

[1] https://github.com/wso2-incubator/testgrid/issues/302